pageserver/src/layered_repository.rs

```diff
@@ -778,6 +788,11 @@ pub struct LayeredTimeline {
     initdb_lsn: Lsn,
 }
 
+struct GcInfo {
+    retain_lsns: Vec<Lsn>,
+    cutoff: Lsn,
+}
```

Comment:
Docstrings pls. Is the cutoff inclusive or exclusive? Possibly useful: https://doc.rust-lang.org/std/ops/struct.RangeTo.html